x-pack/test_serverless/functional/test_suites/common/discover/x_pack/reporting.ts

```diff
@@ -132,25 +124,29 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
       it('generates a report with no data', async () => {
         await PageObjects.reporting.setTimepickerInEcommerceNoDataRange();
+        await PageObjects.discover.clickNewSearchButton();
         await PageObjects.discover.saveSearch('my search - no data - expectReportCanBeCreated');
 
         const res = await getReport();
         expect(res.text).to.be(`\n`);
       });
 
       it('generates a large export', async () => {
+        await PageObjects.discover.clickNewSearchButton();
         const fromTime = 'Apr 27, 2019 @ 23:56:51.374';
         const toTime = 'Aug 23, 2019 @ 16:18:51.821';
         await PageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
-        await PageObjects.discover.clickNewSearchButton();
+        await PageObjects.unifiedFieldList.clickFieldListItemAdd('order_id');
+        await PageObjects.discover.clickFieldSort('order_id', 'Sort A-Z');
+
         await retry.try(async () => {
           expect(await PageObjects.discover.getHitCount()).to.equal('4,675');
         });
         await PageObjects.discover.saveSearch('large export');
 
         // match file length, the beginning and the end of the csv file contents
         const { text: csvFile } = await getReport({ timeout: 80 * 1000 });
-        expect(csvFile.length).to.be(4845684);
+        expect(csvFile.length).to.be(171599);
```

Comment:
To stablize this test we have to add the `order_id` as fields an sort by it. However, now the size of the csv it much smaller, since it no longer contains all the fields. It's still a large report with 4675 records , I'd argue. I think the main purpose here is to export many documents, so I'd consider this change as valid

Reply:
This makes sense, but it's a really big difference in file size and I'm not very familiar with this test. That might be ok, but maybe we should check what @tsullivan thinks first since it looks like he worked on it originally?

One other option that might be worth considering is to create another data view for these tests with a different time field since the e-commerce dataset seems to have multiple date fields. Although if they all have duplicates, we would likely have the same issue. The only alternative I'd see in that case would be to duplicate/modify the dataset to remove duplicates.

Reply:
yes, it's 0.6 MB to 0.02 MB. We could add more fields here to make it larger ... but on the other hand also 0.6 MB ain't really large. But for this tests suite it's the largest number, it apart from the size also start and end of the file are being checked. Another timestamp ain't an option, it seems they are all starting on midnight  `Dec 31, 2016 @ 00:00:00.000`, by coincidence Dec 31th is my birthday 😄 

So we would need other data here, given we want to keep the csv length at approximately the same size 

wdy @tsullivan ?

Reply:
> To stablize this test we have to add the order_id as fields an sort by it. However, now the size of the csv it much smaller, since it no longer contains all the fields.

So originally, the search had no columns selected, and was therefore exporting all fields. Now, you have the `order_id` column selected, so it is not exporting all the fields. Because of that difference, the size of export went down.

> I think the main purpose here is to export many documents

Another purpose of having no fields selected in the sort is to leverage the automatic sorting of `_shard_doc` which is the "tiebreaker" field for sorting when sorting by other fields causes multiple documents to have an equal sort value **[1]**. (See https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html#search-after and search for "tiebreaker"). I understand now that there seems to be no way to test that functionality in E2E tests and have stability. Since that's the case we should rely on unit tests to make sure we are automatically including `_shard_doc` as the tiebreaker. I ran a quick search and it looks like unfortunately, we don't have any unit tests or jest integration tests that are making sure of that. :( 

Anyway, I support making this change for tests stability but perhaps there should be a followup issue to figure out how to test that CSV export is correctly supplying the `_shard_doc` tiebreaker field.

**[1]** Imagine if two documents had the same `order_id` of `X`, our page size is 500, and the first document with `X` as the order ID appeared as the 500th document. If we did not have a tiebreaker, we would mistakenly drop the second document from the export, because we would tell Elasticsearch to begin the 2nd page by searching after where `order_id` is `X`.

Reply:
> [1] Imagine if two documents had the same order_id of X, our page size is 500, and the first document with X as the order ID appeared as the 500th document. If we did not have a tiebreaker, we would mistakenly drop the second document from the export, because we would tell Elasticsearch to begin the 2nd page by searching after where order_id is X.

ok, to keep checking for this situation, we could just leave the test as it was, but don't test for the snapshot, as it could be flaky:

```
expectSnapshot(csvFile.slice(0, 5000)).toMatch();
expectSnapshot(csvFile.slice(-5000)).toMatch();
```

if this is removed, the test ain't flaky anymore. we can also split this test into 2, with and without the order_id.

what I don't understand, in the original test, we sort by timestamp ... then I've been adding order_id, so it's sorting by timestamp and order_id ... don't we use a sorting by tiebreaker also when we sort by timestamp? or is the tiebreaker just added when sorting by 1 fields? thx

Reply:
Thanks for the clarification @tsullivan!

>we can also split this test into 2, with and without the order_id.

Good point and that would work for me 👍

>don't we use a sorting by tiebreaker also when we sort by timestamp? or is the tiebreaker just added when sorting by 1 fields?

I'm not sure I'm understanding correctly, but could it be the case that we always use `_shard_doc` for a tiebreaker, but all `timestamp`/`order_id` combos are unique so we never have to fall back to it when sorting in that scenario (and therefore never hit the inconsistent `_shard_doc` issue in serverless)?

Reply:
I've applied the split of the test cases in 783b86c7b791a91fe5736be3d10a8c5f063f1320, so we now have best of both worlds 

> I'm not sure I'm understanding correctly, but could it be the case that we always use _shard_doc for a tiebreaker, but all timestamp/order_id combos are unique so we never have to fall back to it when sorting in that scenario (and therefore never hit the inconsistent _shard_doc issue in serverless)?

Yes this makes sense 👍 , anyway, it's covered now

Reply:
> could it be the case that we always use _shard_doc for a tiebreaker

This is the case, we can see this in [the code](https://github.com/elastic/kibana/blob/bf40e560e793457eed0eb9cd72f2308a4028c5bd/packages/kbn-generate-csv/src/lib/search_cursor_pit.ts#L159-L160). We are using the unmodified `sort` field from the final document in the page of results, which includes `_shard_doc` automatically from Elasticsearch.